deployment/streaming/scripts/emr-configurations/batch-process.json

```diff
@@ -0,0 +1,93 @@
+[
+  {
+    "Classification": "spark",
+    "Properties": {
+      "maximizeResourceAllocation": "false"
+    }
+  },
+  {
+    "Classification": "spark-defaults",
+    "Properties": {
+        "spark.driver.maxResultSize": "3G",
+        "spark.dynamicAllocation.enabled": "true",
+        "spark.shuffle.service.enabled": "true",
+        "spark.shuffle.compress": "true",
+        "spark.shuffle.spill.compress": "true",
+        "spark.sql.shuffle.partitions": "2000",
+        "spark.speculation": "true",
+        "spark.rdd.compress": "true",
+        "spark.executor.memory": "2G",
+        "spark.executor.memoryOverhead": "1G",
+        "spark.driver.cores": "2",
+        "spark.driver.memory": "10G",
+        "spark.driver.memoryOverhead": "1G",
+        "spark.driver.maxResultSize": "3G",
+        "spark.executor.extraJavaOptions" : "-XX:+UseParallelGC -Dgeotrellis.s3.threads.rdd.write=64"
```

Comment:
We are cargo culting the shit out of this file.  Is this the time where we pair it down to essentials?  For example, I think that `spark.executor.memory` and `spark.driver.memory` might be rendered useless when `spark.dynamicAllocation.enabled` is set.  There's also the duplicae `spark.driver.maxResultSize`.  Many of the other settings below may be equally needless?  It's hard to tell since only the one who originally wrote this configuration knows what is truly necessary and what just felt right to include at the time.

I realize it's not fair to come down on this PR for the sins of its fathers, but then, maybe you shouldn't have copied it in!  :rofl:

Reply:
Yeah you got me. We gotta do it eventually and since we're in a quiet period where we can kill staging a few times, I can give reducing this to some much smaller config a shot over the next few days. 

With that said, I'm not super familiar with a number of these parameters, so if you'd be willing to submit a diff or PR to this branch that deletes / edits a bunch of these that you're relatively confident in changing, that would be super helpful as a starting point.

Reply:
I think this will be of use: https://docs.aws.amazon.com/emr/latest/ReleaseGuide/emr-spark-configure.html

No guarantees about how useful, though.

Reply:
> if you'd be willing to submit a diff or PR to this branch that deletes / edits a bunch of these that you're relatively confident in changing

Just wanted to highlight this again. I don't feel confident enough within the bounds of this task time allotment to go changing/deleting a number of these spark variables given the long debug cycle time (which I probably should have made more clear in my original response). If you feel the same way, then we should defer this to a separate issue since we have a known good working configuration now.

Reply:
I think it's reasonable to bump into the future.

Reply:
👍 created https://github.com/azavea/osmesa/issues/188